IntegrationTests/Services/AWSSTSIntegrationTests/STSWebIdentityCredentialsProviderTests.swift

```diff
@@ -0,0 +1,206 @@
+        _ = try await iamClient.putRolePolicy(input: PutRolePolicyInput(
+            policyDocument: roleIdentityPolicy,
+            policyName: identityPolicyName,
+            roleName: roleName
+        ))
+
+        // Create the Cognito identity pool that allows unauthenticated identities
+        cognitoIdentityClient = try CognitoIdentityClient(region: region)
+        let identityPool = try await cognitoIdentityClient.createIdentityPool(
+            input: CreateIdentityPoolInput(
+                allowUnauthenticatedIdentities: true,
+                identityPoolName: identityPoolName
+        ))
+        identityPoolId = identityPool.identityPoolId
+
+        // Get Cognito ID from Cognito identity pool
+        stsClient = try STSClient(region: region)
+        let accountId = try await stsClient.getCallerIdentity(input: GetCallerIdentityInput()).account
+        let cognitoId = try await cognitoIdentityClient.getId(input: GetIdInput(
+            accountId: accountId, identityPoolId: identityPoolId
+        ))
+
+        // Get OIDC token from Cognito identity pool using Cognito ID
+        oidcToken  = try await cognitoIdentityClient.getOpenIdToken(
+            input: GetOpenIdTokenInput(identityId: cognitoId.identityId)
+        ).token
+        // Save OIDC token to a file then save filepath
+        try saveTokenIntoFile()
+
+        // Construct STS web identity credentials provider
+        let webIdentityCredentialsProvider = try STSWebIdentityCredentialsProvider(
+            region: region,
+            roleArn: roleArn,
+            roleSessionName: roleSessionName,
+            tokenFilePath: tokenFilePath
+        )
+
+        // Construct STS config with STS web identity credentials provider
+        stsConfig = try await STSClient.STSClientConfiguration(
+            credentialsProvider: webIdentityCredentialsProvider,
+            region: region
+        )
+
+        // Construct STS client with just the...
+        // ...STS web identity credentials provider configured
+        webIdentityStsClient = STSClient(config: stsConfig)
+    }
+
+    override func tearDown() async throws {
+        // Delete inline identity policy of the role
+        let policyName = try await iamClient.listRolePolicies(input: ListRolePoliciesInput(roleName: roleName)).policyNames
+        _ = try await iamClient.deleteRolePolicy(input: DeleteRolePolicyInput(
+            policyName: policyName?[0], roleName: roleName
+        ))
+        // Delete role
+        _ = try await iamClient.deleteRole(input: DeleteRoleInput(roleName: roleName))
+        // Delete Cognito identity pool
+        _ = try await cognitoIdentityClient.deleteIdentityPool(input: DeleteIdentityPoolInput(identityPoolId: identityPoolId))
+        // Delete token file
+        try deleteTokenFile()
+    }
+
+    // Helper methods for saving & deleting token file
+    private func saveTokenIntoFile() throws {
+        tokenFilePath = NSHomeDirectory() + "/token.txt"
```

Comment:
We might want to use `FileManager.default.temporaryDirectory` instead of NSHomeDirectory()

Reply:
Changed as suggested.